### PR TITLE
Add mimir enabled flag

### PR DESCRIPTION
**Pull request by QuentinBisson**
Before adding a new alerting rule into this repository you should consider creating an SLO rules instead.
SLO helps you both increase the quality of your monitoring and reduce the alert noise.

* How to create a SLO rule: https://github.com/giantswarm/sloth-rules#how-to-create-a-slo
* Documentation: https://intranet.giantswarm.io/docs/monitoring/slo-alerting/

---
Towards: https://github.com/giantswarm/roadmap/issues/3157

This PR adds the mimir.enabled flag for a preparation to mimir migration and do a whole bunch of really small fixes

### Checklist

- [x] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/aws.management-cluster.rules.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+    # Kept because it will be gone with Vintage
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/aws.management-cluster.rules.yml:8]:
```suggestion
    # No need for .Values.mimir.enabled condition - will be gone with Vintage
```

File: helm/prometheus-rules/templates/alerting-rules/aws.workload-cluster.rules.yml [hunk 2]
```diff
@@ -5,6 +5,7 @@ metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+    # Kept because it will be gone with Vintage
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/aws.workload-cluster.rules.yml:8]:
```suggestion
    # No need for .Values.mimir.enabled condition - will be gone with Vintage
```

File: helm/prometheus-rules/templates/alerting-rules/bastions.rules.yml [hunk 3]
```diff
@@ -5,6 +5,7 @@ metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+    # Kept because it will be gone with Vintage
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/bastions.rules.yml:8]:
```suggestion
    # No need for .Values.mimir.enabled condition - will be gone with Vintage
```

File: helm/prometheus-rules/templates/alerting-rules/cluster-service.rules.yml [hunk 4]
```diff
@@ -5,6 +5,7 @@ metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+    # Kept because it will be gone with Vintage
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/cluster-service.rules.yml:8]:
```suggestion
    # No need for .Values.mimir.enabled condition - will be gone with Vintage
```

File: helm/prometheus-rules/templates/alerting-rules/crsync.rules.yml [hunk 5]
```diff
@@ -5,6 +5,7 @@ metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+    # Kept because it will be gone with Gorilla
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/crsync.rules.yml:8]:
```suggestion
    # No need for .Values.mimir.enabled condition - will be gone with Vintage
```

File: helm/prometheus-rules/templates/alerting-rules/elasticsearch.rules.yml [hunk 6]
```diff
@@ -5,6 +5,7 @@ metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+    # Kept because it will be gone with Vintage
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/elasticsearch.rules.yml:8]:
```suggestion
    # No need for .Values.mimir.enabled condition - will be gone with Vintage
```

File: helm/prometheus-rules/templates/alerting-rules/kiam.rules.yml [hunk 7]
```diff
@@ -5,6 +5,7 @@ metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+    # Kept because it will be gone with Vintage
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/kiam.rules.yml:8]:
```suggestion
    # No need for .Values.mimir.enabled condition - will be gone with Vintage
```

File: helm/prometheus-rules/templates/alerting-rules/kvm.management-cluster.rules.yml [hunk 8]
```diff
@@ -5,6 +5,7 @@ metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+    # Kept because it will be gone with Vintage
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/kvm.management-cluster.rules.yml:8]:
```suggestion
    # No need for .Values.mimir.enabled condition - will be gone with Vintage
```

File: helm/prometheus-rules/templates/alerting-rules/kvm.workload-cluster.rules.yml [hunk 9]
```diff
@@ -5,6 +5,7 @@ metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+    # Kept because it will be gone with Vintage
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/kvm.workload-cluster.rules.yml:8]:
```suggestion
    # No need for .Values.mimir.enabled condition - will be gone with Vintage
```

File: helm/prometheus-rules/templates/alerting-rules/managed-logging.rules.yml [hunk 10]
```diff
@@ -5,6 +5,7 @@ metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+    # Kept because it will be gone with Vintage
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/managed-logging.rules.yml:8]:
```suggestion
    # No need for .Values.mimir.enabled condition - will be gone with Vintage
```

File: helm/prometheus-rules/templates/alerting-rules/release.rules.yml [hunk 11]
```diff
@@ -5,6 +5,7 @@ metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+    # Kept because it will be gone with Vintage
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/release.rules.yml:8]:
```suggestion
    # No need for .Values.mimir.enabled condition - will be gone with Vintage
```

File: helm/prometheus-rules/templates/alerting-rules/vault.rules.yml [hunk 12]
```diff
@@ -5,6 +5,7 @@ metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+    # Kept because it will be gone with Vintage
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/vault.rules.yml:8]:
```suggestion
    # No need for .Values.mimir.enabled condition - will be gone with Vintage
```

**Review by hervenicol — COMMENTED** [review]
The comment makes sense when reading the PR, but may not make sense when reading the files in a few months / weeks.

File: helm/prometheus-rules/templates/alerting-rules/inhibit.management-cluster.rules.yml [hunk 13]
```diff
@@ -1,10 +1,13 @@
+{{- if eq .Values.managementCluster.provider.flavor "vintage" }}
 apiVersion: monitoring.coreos.com/v1
 kind: PrometheusRule
 metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+{{- if not .Values.mimir.enabled }}
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/inhibit.management-cluster.rules.yml:8]: This file is only for vintage clusters; I guess we don't need the condition then?

File: helm/prometheus-rules/templates/alerting-rules/credentiald.rules.yml [hunk 14]
```diff
@@ -41,3 +43,4 @@ spec:
         severity: notify
         team: phoenix
         topic: orchestration
+{{- end }}
```
Comment (QuantumEnigmaa) [inline helm/prometheus-rules/templates/alerting-rules/credentiald.rules.yml:46]:
```suggestion
{{- end }}

```

File: helm/prometheus-rules/templates/alerting-rules/inhibit.management-cluster.rules.yml (hunk 13 above)
Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alerting-rules/inhibit.management-cluster.rules.yml:8]: Well that's true but i'm not sure if phoenix will not add CAPI ones here

File: helm/prometheus-rules/templates/alerting-rules/credentiald.rules.yml [hunk 15]
```diff
@@ -1,9 +1,11 @@
+{{- if eq .Values.managementCluster.provider.flavor "vintage" }}
 apiVersion: monitoring.coreos.com/v1
 kind: PrometheusRule
 metadata:
   creationTimestamp: null
   labels:
     {{- include "labels.common" . | nindent 4 }}
+    # Kept because it will be gone with Vintage
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/alerting-rules/credentiald.rules.yml:8]: Oops, I had missed one!
```suggestion
    # No need for .Values.mimir.enabled condition - will be gone with Vintage
```

**Review by hervenicol — APPROVED** [review]
LGTM.
It could be nice to link to this issue when doing the changes in our `config`.

**Comment by QuentinBisson** [comment]
Linked in this https://github.com/giantswarm/shared-configs/pull/8

File: helm/prometheus-rules/templates/alerting-rules/inhibit.management-cluster.rules.yml [hunk 16]
```diff
@@ -105,3 +108,4 @@ spec:
         team: phoenix
         topic: status
     {{- end }}
+{{- end }}
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/inhibit.management-cluster.rules.yml:111]:
```suggestion
{{- end }}

```

**Pull request merged**